src/meshes/p4est_mesh_view.jl

```diff
@@ -0,0 +1,209 @@
+    if get(io, :compact, false)
+        show(io, mesh)
+    else
+        setup = [
+            "#trees" => ntrees(mesh),
+            "current #cells" => ncellsglobal(mesh),
+            "polydeg" => length(mesh.nodes) - 1,
+        ]
+        summary_box(io,
+                    "P4estMeshView{" * string(ndims(mesh)) * ", " * string(real(mesh)) *
+                    "}", setup)
+    end
+end
+
+function balance!(mesh::P4estMeshView{2}, init_fn = C_NULL)
+    p4est_balance(mesh.p4est, P4EST_CONNECT_FACE, init_fn)
+    # Due to a bug in `p4est`, the forest needs to be rebalanced twice sometimes
+    # See https://github.com/cburstedde/p4est/issues/112
+    p4est_balance(mesh.p4est, P4EST_CONNECT_FACE, init_fn)
+end
+
+@inline ncells(mesh::P4estMeshView) = Int(mesh.p4est.local_num_quadrants[])
+
+function calc_node_coordinates!(node_coordinates,
+                                mesh::P4estMeshView{2},
+                                nodes::AbstractVector)
+    # We use `StrideArray`s here since these buffers are used in performance-critical
+    # places and the additional information passed to the compiler makes them faster
+    # than native `Array`s.
+    tmp1 = StrideArray(undef, real(mesh),
+                       StaticInt(2), static_length(nodes), static_length(mesh.nodes))
+    matrix1 = StrideArray(undef, real(mesh),
+                          static_length(nodes), static_length(mesh.nodes))
+    matrix2 = similar(matrix1)
+    baryweights_in = barycentric_weights(mesh.nodes)
+
+    # Macros from `p4est`
+    p4est_root_len = 1 << P4EST_MAXLEVEL
+    p4est_quadrant_len(l) = 1 << (P4EST_MAXLEVEL - l)
+
+    trees = unsafe_wrap_sc(p4est_tree_t, mesh.p4est.trees)
+
+    for tree in eachindex(trees)
+        offset = trees[tree].quadrants_offset
+        quadrants = unsafe_wrap_sc(p4est_quadrant_t, trees[tree].quadrants)
+
+        for i in eachindex(quadrants)
+            element = offset + i
+            quad = quadrants[i]
+
+            quad_length = p4est_quadrant_len(quad.level) / p4est_root_len
+
+            nodes_out_x = 2 * (quad_length * 1 / 2 * (nodes .+ 1) .+
+                           quad.x / p4est_root_len) .- 1
+            nodes_out_y = 2 * (quad_length * 1 / 2 * (nodes .+ 1) .+
+                           quad.y / p4est_root_len) .- 1
+            polynomial_interpolation_matrix!(matrix1, mesh.nodes, nodes_out_x,
+                                             baryweights_in)
+            polynomial_interpolation_matrix!(matrix2, mesh.nodes, nodes_out_y,
+                                             baryweights_in)
+
+            multiply_dimensionwise!(view(node_coordinates, :, :, :, element),
+                                    matrix1, matrix2,
+                                    view(mesh.tree_node_coordinates, :, :, :, tree),
+                                    tmp1)
+        end
+    end
+
+    return node_coordinates
+end
+
+function save_mesh_file(mesh::P4estMeshView, output_directory; timestep = 0,
```

Comment:
Couple of questions:
Is this routine called for every `P4estMeshView` instance? Does that mean the file is written multiple times?
Where is the information stored what section of the whole mesh is only the view's part? Is it possible to load a `P4estMeshView` mesh?

Reply:
This function is called for every system. The system is specified via the `system` parameter. The resulting files are of the same format as the p4est files and can be loaded the same way. This is the same approach we have for the structured mesh views.

Reply:
Do I understand that right, that every system, respectively, MeshView shares the same forest? Why write out the identical mesh information multiple times? Can the mesh info file not be shared by all systems resp. MeshViews?

Reply:
The MeshViews point to different parts of the parent forest. The mesh information that is written into the different mesh files is not shared.  For instance, we would have different `tree_node_coordinates`. Therefore, the mesh info file cannot be shared.